Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -79,11 +77,23 @@ private ValidationResult Validate()
 
                         locationStack.Push(subjectReader.LocalName);
                         validationResult = ValidateAttributes();
+
                         if (subjectReader.IsEmptyElement)
                         {
+                            // TODO Why? Please help to explain
```

Comment:
I'm not sure why this is done.
Can you help please to give a good comment?

Reply:
Is it used to detect cyclic references or something?

Reply:
Thought: could it be a leftover after I added `reader.MoveToElement();` to `GetAttributes`in #1170?

Reply:
Once again I tried to understand the block.
I think, I understood.
Please check my comment whether you agree.